src/function/scalar/sequence/nextval.cpp

```diff
@@ -124,7 +124,8 @@ static unique_ptr<FunctionData> NextValBind(ClientContext &context, ScalarFuncti
 		if (!seqname.IsNull()) {
 			D_ASSERT(seqname.type().id() == LogicalTypeId::VARCHAR);
 			auto qname = QualifiedName::Parse(StringValue::Get(seqname));
-			sequence = Catalog::GetCatalog(context).GetEntry<SequenceCatalogEntry>(context, qname.schema, qname.name);
+			sequence = Catalog::GetCatalog(context).GetEntry<SequenceCatalogEntry>(context, qname.schema, qname.name,
```

Comment:
This is where I played things fast and loose to make this work: before, this would throw an error if the sequence that the `nextval` function was expecting didn't exist, but since the impl of `nextval` already also supports the potential for its argument to not be a foldable expression, I could essentially cheat here and have this block of code set a `nullptr` for the `sequence` argument in case that the sequence that `nextval` will be calling doesn't exist at the time the function is bound.

Reply:
Indeed ideally this is removed - particularly because if `sequence` is not set many catalog lookups will be performed, which will make insertions much slower.

Reply:
So the bit I haven't figured out how to pull off is this: I'm defining the sequence and the `nextval` expression to use it as the default value for the ColumnDefinition during parsing, but the `nextval` function will be bound (and thus have its sequence catalog entry set) _before_ I am actually able to create the sequence in the catalog during table creation. Being lazy here (i.e., allowing for the catalog entry to not exist at the time of function binding) lets me get around this, but as you said, the performance hit of having to do this lookup over and over again while doing lots of inserts could well be substantial.

Is there a good way out of this puzzle?